esp/src/eclwatch/DynamicESDLDetailsWidget.js

```diff
@@ -0,0 +1,313 @@
+                                        Severity: "Message",
+                                        Source: "WsESDLConfig.DeleteESDLBinding",
+                                        Exceptions: [{ Source: context.i18n.DeletedBinding, Message: context.i18n.BindingDeleted }]
+                                    });
+                                    context.addBindingButton.set("disabled", false);
+                                    context.deleteBindingButton.set("disabled", true);
+                                    context.widget._Binding.set("disabled", true);
+                                    context.params.Owner._onRefresh({});
+                                    context.params.Owner.grid.deselect(context.params.__hpcc_id);
+                                    context.params.Owner.grid.select(context.params.__hpcc_id);
+                                }
+                            }
+                        });
+                    }
+                },
+                label: context.i18n.DeleteBinding
+            }).placeAt(this.addBindingButton.domNode, "after");
+            this.definitionDropDown = new TargetSelectWidget({});
+            this.definitionDropDown.init({
+                LoadDESDLDefinitions: true
+            });
+            this.dialog = new Dialog({
+                title: context.i18n.PleasePickADefinition,
+                style: "width: 300px;"
+            });
+            this.dialog.addChild(this.definitionDropDown);
+            this.addDefinitionButton = new Button({
+                disabled: false,
+                style: "float:right;",
+                label: "Apply",
+                onClick: function (val) {
+                    WsESDLConfig.GetESDLDefinition({
+                        request: {
+                            Id: context.definitionDropDown.get("value"),
+                            ReportMethodsAvailable: true
+                        }
+                    }).then(function (response) {
+                        if (response.GetESDLDefinitionResponse.status.Code === 0) {
+                             WsESDLConfig.PublishESDLBinding({
+                                request: {
+                                    EspProcName: context.params.__hpcc_parentName,
+                                    EspBindingName: context.params.Name,
+                                    EsdlDefinitionID: response.GetESDLDefinitionResponse.Id,
+                                    Overwrite: true,
+                                    EsdlServiceName: response.GetESDLDefinitionResponse.ESDLServices.Name,
+                                    Config: "<Methods><Method name='" + response.GetESDLDefinitionResponse.Methods.Method[0].Name + "'/></Methods>"
+                                }
+                            }).then(function (publishresponse) {
+                                if (publishresponse.PublishESDLBindingResponse.status.Code === 0) {
```

Comment:
Does this need an "exists" test before using it (there might have been an exception)?